src/Fixer/PhpUnit/PhpUnitMethodCasingFixer.php

```diff
@@ -0,0 +1,277 @@
+            if (!$this->isTestMethod($tokens, $index)) {
+                continue;
+            }
+
+            $functionNameIndex = $tokens->getNextMeaningfulToken($index);
+            $functionName = $tokens[$functionNameIndex]->getContent();
+            $newFunctionName = $this->updateMethodCasing($functionName);
+
+            if ($newFunctionName !== $functionName) {
+                $tokens[$functionNameIndex] = new Token([T_STRING, $newFunctionName]);
+            }
+
+            $docBlockIndex = $this->getDocBlockIndex($tokens, $index);
+            if ($this->hasDocBlock($tokens, $index)) {
+                $this->updateDocBlock($tokens, $docBlockIndex);
+            }
+        }
+    }
+
+    /**
+     * @param string $functionName
+     *
+     * @return string
+     */
+    private function updateMethodCasing($functionName)
+    {
+        if (self::CAMEL_CASE === $this->configuration['case']) {
+            $newFunctionName = $functionName;
+            $newFunctionName = str_replace('_', ' ', $newFunctionName);
+            $newFunctionName = ucwords($newFunctionName);
+            $newFunctionName = str_replace(' ', '', $newFunctionName);
+            $newFunctionName = lcfirst($newFunctionName);
+        } else {
+            $newFunctionName = Utils::camelCaseToUnderscore($functionName);
+        }
+
+        return $newFunctionName;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int$index
+     *
+     * @return bool
+     */
+    private function isTestMethod(Tokens $tokens, $index)
+    {
+        // Check if we are dealing with a (non abstract, non lambda) function
+        if (!$this->isMethod($tokens, $index)) {
+            return false;
+        }
+
+        // if the function name starts with test its a test
+        $functionNameIndex = $tokens->getNextMeaningfulToken($index);
+        $functionName = $tokens[$functionNameIndex]->getContent();
+
+        if ($this->startsWith('test', $functionName)) {
+            return true;
+        }
+        // If the function doesn't have test in its name, and no doc block, its not a test
+        if (!$this->hasDocBlock($tokens, $index)) {
+            return false;
+        }
+
+        $docBlockIndex = $this->getDocBlockIndex($tokens, $index);
+        $doc = $tokens[$docBlockIndex]->getContent();
+        if (false === strpos($doc, '@test')) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    private function isMethod(Tokens $tokens, $index)
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        return $tokens[$index]->isGivenKind(T_FUNCTION) && !$tokensAnalyzer->isLambda($index);
+    }
+
+    /**
+     * @param string $needle
+     * @param string $haystack
+     *
+     * @return bool
+     */
+    private function startsWith($needle, $haystack)
+    {
+        $len = strlen($needle);
```

Comment:
(Personal) I'd drop `$len` here and put `strlen` directly in `substr` to have one-liner.